fabric-entity-events-v1/src/main/java/net/fabricmc/fabric/api/entity/event/v1/ServerPlayerEvents.java

```diff
@@ -45,6 +47,30 @@
 		}
 	});
 
+	/**
+	 * An event that is called when a player takes fatal damage.
+	 *
+	 * <p>Mods can cancel this to keep the player alive.
+	 *
+	 * <p>Vanilla checks for player health <= 0 each frame (with {@link LivingEntity#isDead()}), and kills if true -
+	 * so the player will die next frame if this is cancelled. It's assumed that the listener will do something to
+	 * prevent this, for example:
+	 *
+	 * <li>
+	 *     <ul>a minigame mod teleporting the player into a 'respawn room' and setting their health to 20.0</ul>
+	 *     <ul>a mod that changes death mechanics switching the player over to the mod's play-mode, where death doesn't
+	 *     apply</ul>
+	 * </li>
+	 */
+	public static final Event<CancelDeath> CANCEL_DEATH = EventFactory.createArrayBacked(CancelDeath.class, callbacks -> (oldPlayer, damageSource, damageAmount) -> {
+		for (CancelDeath callback : callbacks) {
+			if (callback.cancelDeath(oldPlayer, damageSource, damageAmount)) {
```

Comment:
Imo I'd change `oldPlayer` to just `player` as that's what is used in the event parameters.

```suggestion
	public static final Event<CancelDeath> CANCEL_DEATH = EventFactory.createArrayBacked(CancelDeath.class, callbacks -> (player, damageSource, damageAmount) -> {
		for (CancelDeath callback : callbacks) {
			if (callback.cancelDeath(player, damageSource, damageAmount)) {
```

Reply:
That's what I get for copy-pasting the copyplayerevent :sweat_smile: